File: Fourth Semester - Design and Programming Environments/Java/ProiectJava/java/org/example/modul/validator/AgentValidator.java
package org.example.modul.validator;


import org.example.exception.ValidationException;
import org.example.modul.Agent;

public class AgentValidator implements Validator<Agent> {
    @Override
    public void validate(Agent entity) throws ValidationException {
        if (entity == null) {
            throw new IllegalArgumentException("Agency is null");
        }
        if (entity.getName() == null || entity.getName().isEmpty()) {
            throw new ValidationException("Agency name is null or empty");
        }
        if (entity.getUsername() == null || entity.getUsername().isEmpty()){
            throw new ValidationException("Agency username is null or empty");
        }
        if (entity.getPassword() == null || entity.getPassword().isEmpty()){
            throw new ValidationException("Agency password is null or empty");
        }
    }
}

File: Fourth Semester - Design and Programming Environments/C#/ProiectC#Networking/TravelModel/model/validator/AgentValidator.cs
namespace TravelModel.model.validator;

public class AgentValidator : IValidator<Agent>
{
    public bool Validate(Agent entity)
    {
        if (entity == null)
        {
            throw new ArgumentNullException(nameof(entity), "Agency cannot be null");
        }

        if (string.IsNullOrEmpty(entity.Name))
        {
            throw new ValidationException("Agent name cannot be empty");
        }

        if (string.IsNullOrEmpty(entity.Username))
        {
            throw new ValidationException("Agency username cannot be empty");
        }

        if (string.IsNullOrEmpty(entity.Password))
        {
            throw new ValidationException("Agency password cannot be empty");
        }
        
        return true;
    }
}